### PR TITLE
[Merged by Bors] - feat: port Rat.cast positivity extension

### PR DESCRIPTION
**Pull request by dwrensha**
See the mathlib3 version here: https://github.com/leanprover-community/mathlib/blob/3365b20c2ffa7c35e47e5209b89ba9abdddf3ffe/src/tactic/positivity.lean#L737-L741

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Positivity/Basic.lean [hunk 1]
```diff
@@ -460,6 +463,25 @@ def evalIntCast : PositivityExt where eval {u α} _zα _pα e := do
   | .none =>
     pure .none
 
+/-- Extension for Rat.cast. -/
+@[positivity Rat.cast _]
+def evalRatCast : PositivityExt where eval {u α} _zα _pα e := do
+  let (.app _ (a : Q(ℚ))) ← whnfR e | throwError "not Rat.cast"
```
Comment (dwrensha) [inline Mathlib/Tactic/Positivity/Basic.lean:469]: I wanted to make this a `~q` match, like this
```
def evalRatCast : PositivityExt where eval {u} (α : Q(Type u)) _zα _pα (e : Q($α)) := do
  let _rα ← synthInstanceQ (q(RatCast $α))
  let ~q(Rat.cast ($a : ℚ)) := e | throwError "failed to match Rat.cast"
...
```
but I couldn't figure out how to make it work.

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

File: Mathlib/Tactic/Positivity/Basic.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Tactic/Positivity/Basic.lean]: Qq needs explicit type annotations on most `let`s

Comment (dwrensha) [reply Mathlib/Tactic/Positivity/Basic.lean]: Thank you!

**Pull request closed without merging**